macosx/TorrentCellActionButton.mm

```diff
@@ -0,0 +1,90 @@
+// This file Copyright © 2006-2023 Transmission authors and contributors.
+// It may be used under the MIT (SPDX: MIT) license.
+// License text can be found in the licenses/ folder.
+
+#import "TorrentCellActionButton.h"
+#import "TorrentTableView.h"
+#import "Torrent.h"
+
+@interface TorrentCellActionButton ()
+@property(nonatomic) NSTrackingArea* fTrackingArea;
+@property(nonatomic) NSImage* fImg;
+@property(nonatomic) NSImage* fAltImg;
+@property(nonatomic) TorrentTableView* torrentTableView;
+@property(nonatomic, readonly) NSUserDefaults* fDefaults;
+@end
+
+@implementation TorrentCellActionButton
+
+- (void)awakeFromNib
+{
+    _fDefaults = NSUserDefaults.standardUserDefaults;
+    self.fImg = self.image;
+
+    // hide image by default and show only on hover
+    self.fAltImg = [[NSImage alloc] init];
+    self.image = self.fAltImg;
+
+    // disable button click highlighting
+    [self.cell setHighlightsBy:NSNoCellMask];
+}
+
+- (void)display
+{
+    [super display];
+}
+
+- (void)initTorrentTableView
+{
+    if (!self.torrentTableView)
+    {
+        self.torrentTableView = (TorrentTableView*)[[[self superview] superview] superview];
```

Comment:
Since there is an `awakeFromView` in this class, can't we link the tableView in the xib instead of exploring the superviews?

Reply:
I just tried this and don't ask me why, but the connection resolvers for the first cell.... but none other...

Reply:
perhaps there's a better way, for instance in TorrentCell, and SmallTorrentCell I added a property which I passed the TorrentTableView too upon creation. Ill take a look

Reply:
same problem with directly connecting the reference in code... In fact its actually worse, TorrentTableView is always nil
🤮